### PR TITLE
Update README with details about design doc repo

### PR DESCRIPTION
**Pull request by nickjalbert**
Addresses [Issue 78: Add design doc process to documentation](https://github.com/agentos-project/agentos/issues/78).

The [PR](https://github.com/agentos-project/design_docs/pull/1) in the `design_docs` repo adds much more detail its README about the process. See [here](https://github.com/agentos-project/design_docs/blob/00e28049aa46797e55be76c202849c41aeac34ce/README.rst).

### CODE REVIEW

**Comment by andyk** [comment]
This looks good!

Should we also mention that we use the GitHub issue tracker and provide guidance for what types of features require a design doc vs issue or both?

**Comment by andyk** [comment]
Also, what do you think about moving this new section up higher. Maybe a before the section on building the docs?

**Review by andyk — APPROVED** [review]

**Comment by andyk** [comment]
Looking good.

* looks like you need to also push updated readme
* Your internal link that uses `:ref:...` will be broken till we fix #67 but I don’t think we should block this on that.

**Comment by nickjalbert** [comment]
@andyk looks like I figured out how to get internal linking working.  See if it works for you.

Also, added a blurb about the general development process. Not sure if you read it, but tell me if you want any changes!

**Comment by nickjalbert** [comment]
Ah, looks like my solution only works for links in the same document and does not fix https://github.com/agentos-project/agentos/issues/67.

**Comment by nickjalbert** [comment]
Will merge after tests are green.

**Comment by andyk** [comment]
@nickjalbert I noticed that merging this did not automatically close #78. I think if you use the "Fixes" or "Closes" keyword in a commit message that is part of a PR (e.g., "fixes #78"), then GitHub will automatically link the PR to an Issue and close it when the PR is merged (see the references subsection in [these github docs](https://guides.github.com/features/issues/?#notifications)).

I think you can also manually link Issues to PRs using the "Linked Issues" metadata field of an Issue.

**Comment by nickjalbert** [comment]
Ah, good to know.  Thanks!

**Comment by andyk** [comment]
> 
> 
> Ah, good to know. Thanks!

Also, just found [this documentation](https://docs.github.com/en/github/writing-on-github/autolinked-references-and-urls#issues-and-pull-requests) which covers the syntax for linking to stuff in a diff repo: it's org_name/repo_name#NUM, e.g.: agentos-project/design_docs#5 should link to the discussion for the core abstractions design doc and agentos-project/design_docs#3 should link to the PR for that same design doc.

**Pull request merged**
